File: app/src/main/java/example/simple/itemdecor/grid/GridItemBinder.java
package example.simple.itemdecor.grid;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import me.drakeet.multitype.ItemViewBinder;
import me.simple.itemdecor.demo.R;

public class GridItemBinder extends ItemViewBinder<GridItemBean, GridItemBinder.VH> {

    @NonNull
    @Override
    protected VH onCreateViewHolder(@NonNull LayoutInflater inflater, @NonNull ViewGroup parent) {
        return new VH(inflater.inflate(R.layout.item_grid, parent, false));
    }

    @Override
    protected void onBindViewHolder(@NonNull final VH holder, @NonNull GridItemBean item) {
        holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                getAdapter().getItems().remove(holder.getAdapterPosition());
                getAdapter().notifyItemRemoved(holder.getAdapterPosition());
                return true;
            }
        });
    }

    class VH extends RecyclerView.ViewHolder {

        public VH(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/example/simple/itemdecor/grid/GridItemBinder.kt
package example.simple.itemdecor.grid

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import example.simple.itemdecor.R
import me.drakeet.multitype.ItemViewBinder

class GridItemBinder : ItemViewBinder<GridItemBean, GridItemBinder.VH>() {
    override fun onCreateViewHolder(inflater: LayoutInflater, parent: ViewGroup): VH {
        return VH(inflater.inflate(R.layout.item_grid, parent, false))
    }

    override fun onBindViewHolder(holder: VH, item: GridItemBean) {
        holder.itemView.setOnLongClickListener {
            adapter.items.removeAt(holder.adapterPosition)
            adapter.notifyItemRemoved(holder.adapterPosition)
            true
        }
    }

    inner class VH(itemView: View) : ViewHolder(itemView)
}